### PR TITLE
Panning

### PR DESCRIPTION
**Pull request by daviddq**
Move .nodes-holder instead of moving all nodes.

### CODE REVIEW

File: src/js/classes/node.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-var globalNodeIndex = 0;
+let globalNodeIndex = 0;
 const NodeExpandWidth = 300;
 const NodeExpandHeight = 150;
 const ClipNodeTextLength = 1024;
 const bbcode = require('bbcode');
 
-export var Node = function(options = {}) {
-  var self = this;
+export let Node = function(options = {}) {
```
Comment (blurymind) [inline src/js/classes/node.js:10]: wondering now if this should be a const :thinking:

**Comment by blurymind** [comment]
Nice work so far!
This seems to have broken the zoom level buttons in the bottom right of the app

**Comment by blurymind** [comment]
Another problem I notice is that if you right click, the browser's context menu will appear, which is not something we want

File: src/js/classes/workspace.js [hunk 2]
```diff
@@ -18,87 +18,147 @@ export const Workspace = function(app) {
...
+  //
+  // Sets the translation on the transform matrix
+  this.setTranslation = function(x, y, time=0) {
+    self.transform.x = x;
+    self.transform.y = y;
+    self.translate(time);
+  };
+
+  // shiftTranslation
+  //
```
Comment (blurymind) [inline src/js/classes/workspace.js:44]: Do you leave this gap for future TODOs :)
Interesting habit

File: src/js/classes/app.js [hunk 3]
```diff
@@ -119,7 +112,9 @@ export var App = function(name, version) {
       osName = 'mobile';
     }
 
-    if (osName == 'Windows') self.zoomSpeed = 0.1;
+    // Platform specific settings
+    if (osName == 'Windows') self.workspace.zoomSpeed = 0.1;
+    if (osName === 'mobile') self.workspace.zoom(3);
```
Comment (blurymind) [inline src/js/classes/app.js:117]: nit: can be else if

Comment (blurymind) [inline src/js/classes/app.js:117]: Actually, can you remove this line please?
`if (osName === 'mobile') self.workspace.zoom(3);`

It completely breaks yarn on a portrait mode smartphone

Comment (daviddq) [reply src/js/classes/app.js:117]: That line is not new. Are you sure you want to remove it?

Comment (blurymind) [reply src/js/classes/app.js:117]: > That line is not new. Are you sure you want to remove it?

move it back to where it was before

File: src/js/classes/node.js (hunk 1 above)
Comment (daviddq) [reply src/js/classes/node.js:10]: Yes. I did a quick search and replace var -> let

**Comment by daviddq** [comment]
> Also when you pan, are you still altering all node positions or does this fix that problem? Their coordinate stays the same in the file you save - if it hasnt been moved

There's no need to alter the nodes' positions now, so the diff problem should be mitigated.

File: src/js/classes/app.js (hunk 3 above)
Comment (daviddq) [reply src/js/classes/app.js]: All mentioned bugs fixed. Thanks for helping to test it 😉

**Comment by blurymind** [comment]
Nice work on this. :+1:  and thank you for fixing the regressions

The performance boost is noticeable, the codebase- meaner and cleaner, the dif noise that yarn created before when saving files - gone!

merging this now! :tada:

**Pull request merged**